File: i7h.py
import re


def _f(r):
    t = r.group()
    if len(t) <= 2:
        return t
    return f'{t[0]}{len(t)-2}{t[-1]}'


def i18n(s: str) -> str:
    s = re.sub('[A-Za-z]+', _f, s)
    return s

File: i7h.js
export const i18n = (str) => {
  if (typeof str !== "string") {
    throw new Error("Not string");
  }

  return str.replace(/\w+/g, ellipsis);
};

const ellipsis = (word) => {
  if (word.length < 3) {
    return word;
  }

  return `${word.slice(0, 1)}${word.length - 2}${word.slice(-1)}`;
};
